packages/headless/src/features/result-preview/result-preview-slice.ts

```diff
@@ -1,4 +1,5 @@
 import {createReducer} from '@reduxjs/toolkit';
+import {WritableDraft} from 'immer/dist/internal';
```

Comment:
I would not import types from internal of immer, unless absolutely necessary.

```typescript
const resetPreviewContentState = (state:ResultPreviewState) => {

[... ]
```
Should work just fine.